community/ndp/runtime/src/test/java/org/neo4j/ndp/runtime/integration/RecordingCallback.java

```diff
@@ -219,17 +221,17 @@ public String toString()
         }
     }
 
-    private Object[] unwind( RecordStream result ) throws Exception
+    private Record[] unwind( RecordStream result ) throws Exception
     {
-        final List<Object> values = new ArrayList<>();
+        final List<Record> values = new ArrayList<>();
         result.visitAll( new RecordStream.Visitor()
         {
             @Override
             public void visit( Record record )
             {
-                values.add( record.copy() );
+                values.add( new ImmutableRecord( record.fields() ) );
```

Comment:
Since it is test code I didn't really think about memory optimizations. Is that important?